### PR TITLE
Reword bug -> limitation for BaseHTTPMiddleware

### PR DESCRIPTION
**Pull request by adriangb**
Ref https://github.com/encode/starlette/pull/1656#discussion_r911376977

### CODE REVIEW

**Comment by Kludex** [comment]
I mentioned more in the case you wanted to add a note about the "modify request body" thing... 🤔

File: docs/middleware.md [hunk 1]
```diff
@@ -226,9 +226,8 @@ Middleware classes should not modify their state outside of the `__init__` metho
 Instead you should keep any state local to the `dispatch` method, or pass it
 around explicitly, rather than mutating the middleware instance.
 
-!!! bug
-    Currently, the `BaseHTTPMiddleware` has some known issues:
-
+!!! warning
+    Currently, the `BaseHTTPMiddleware` has some known limitations:
```
Comment (Kludex) [inline docs/middleware.md:230]: Without the empty line, the docs "break" 😅 
```suggestion
    Currently, the `BaseHTTPMiddleware` has some known limitations:

```

**Comment by adriangb** [comment]
Oh gotcha, yes that would be good to add! Nonetheless, I do think the wording of "limitation" is better than "bug" for these issues, right?

**Review by Kludex — APPROVED** [review]
Sure! :)

**Pull request merged**